### ISSUE TITLE
Support cross-account send for AWS SES email transport

### ISSUE THREAD

**Issue by mortenmorten** [FlowFuse/flowfuse#4173]
### Description

We are a multi-account AWS user, running self hosted FlowFuse in AWS EKS, and want to send emails using SES on another account than the one running the EKS cluser.

`@aws-sdk/client-ses` supports this by using the [SourceArn](https://github.com/aws/aws-sdk-js-v3/blob/794a37e9795f390d15c9530209e465d099716c86/clients/client-ses/src/models/models_0.ts#L4375) when sending emails.

`nodemailer` supports [custom fields](https://nodemailer.com/transports/ses/#example-1) for the SES transport.

From what I understand all that's required is a new `sourceArn` property in the email config and then something like:
```javascript
if(sesConfig.sourceArn) {
  mailDefaults['ses'] = { SourceArn: sesConfig.sourceArn };
}
```
when creating the SES transport in [postoffice](https://github.com/FlowFuse/flowfuse/blob/main/forge/postoffice/index.js)
 

### Which customers would this be available to

Everyone - CE/Starter/Team/Enterprise

### Have you provided an initial effort estimate for this issue?

I am not a FlowFuse team member

**Comment by hardillb** [comment]
I'll have a look at this.

A bit of searching implies that it needs a little bit more (needs a `FromArn` as well as `SourceArn`

https://github.com/nodemailer/nodemailer/issues/1218

I need to work out how to test this, @mortenmorten do you have any pointers to the AWS doc for how you've set up SES with multiple users?

It will also need a helm chart update to add a way to configure the field

**Comment by hardillb** [comment]
We should be able to test this using localfs

**Comment by mortenmorten** [comment]
@hardillb I used [this AWS blog post](https://aws.amazon.com/blogs/messaging-and-targeting/how-to-grant-another-ses-account-or-user-permission-to-send-emails/) to configure cross-account SES delegation.

TLDR; grant send email access to a role in account A and grant access to the SES identity in role B using trust policy. Then send email setting the ARN of the identity in the SourceArn field from account A.

**Comment by mortenmorten** [comment]
@hardillb Please see the [docs for SendRawEmail](https://docs.aws.amazon.com/ses/latest/APIReference/API_SendRawEmail.html). It states that `FromArn` is optional when using `SourceArn`. `FromArn` will default to `SourceArn` (wrongly written as SourceIdentityArn in the docs).

Would probably be a good idea to add support for setting the `FromArn` anyways.

**Comment by hardillb** [comment]
@mortenmorten I've just tested with the following code (replace with valid identity ARN, region and email addresses):

```
let nodemailer = require("nodemailer");
let aws = require("@aws-sdk/client-ses");
let { defaultProvider } = require("@aws-sdk/credential-provider-node");

const ses = new aws.SES({
  apiVersion: "2010-12-01",
  region: "eu-west-1",
  defaultProvider,
});

// create Nodemailer SES transporter
let transporter = nodemailer.createTransport({
  SES: { ses, aws },
});

// send some mail
transporter.sendMail(
  {
    from: "flowfuse@example.com",
    to: "ben@example.com",
    subject: "Message",
    text: "I hope this message gets sent!",
    ses: {
      SourceArn: 'arn:aws:ses:eu-west-1:xxxxxxxxxxxxx:identity/example.com',
      FromArn: 'arn:aws:ses:eu-west-1:xxxxxxxxxxxxx:identity/example..com'
    },
  },
  (err, info) => {
    if (!err) {
      console.log(info.envelope);
      console.log(info.messageId);
    } else {
      console.log(err)
    }
  }
);

```

It won't send without the `FromArn` being set to the same as the `SourceArn` value.

Can you give that a test locally to see if you see the same please.

needed to install 

```
  "dependencies": {
    "@aws-sdk/client-ses": "^3.614.0",
    "@aws-sdk/credential-provider-node": "^3.614.0",
    "nodemailer": "^6.9.14"
  }
```

**Comment by mortenmorten** [comment]
@hardillb Tested locally with the same results as you. The FromArn is required.

**Comment by hardillb** [comment]
@mortenmorten this is merged, will be included in the 2.7.0 release and there are corresponding additions to the helm chart to support adding the config. https://github.com/FlowFuse/helm/pull/438
